src/tests/performance/get_blocks.rs

```diff
@@ -0,0 +1,188 @@
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     700 │        100 │          0 │         11 │              2 │          1 │          1 │          2 │          3 │          6 │          98.77 │      10.43 │      6629.62 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     800 │        100 │          0 │         11 │              1 │          1 │          1 │          2 │          2 │          5 │          99.02 │       7.62 │     10396.44 │
+    // └─────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴────────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴──────────────┘
+    // *NOTE* run with `cargo test --release  tests::performance::get_blocks -- --nocapture`
+    // Before running test generate dummy devices with different ips using toos/ips.py
+
+    let synth_counts = vec![1, 50, 100, 200, 300, 400, 500, 600, 700, 800];
+
+    let mut table = RequestsTable::default();
+
+    for synth_count in synth_counts {
+        let target = TempDir::new().expect("couldn't create a temporary directory");
+        let mut node = Node::builder()
+            .build(target.path())
+            .expect("unable to build the node");
+        node.start().await;
+
+        let node_addr = node.net_addr().expect("network address not found");
+
+        let mut synth_sockets = Vec::with_capacity(synth_count);
+        let mut ips = IPS.to_vec();
+
+        for _ in 0..synth_count {
+            // If there is address for our thread in the pool we can use it.
+            // Otherwise we'll not set bound_addr and use local IP addr (127.0.0.1).
+            let ip = ips.pop().unwrap_or("127.0.0.1");
+
+            let ip = SocketAddr::new(IpAddr::V4(Ipv4Addr::from_str(ip).unwrap()), 0);
+            let socket = TcpSocket::new_v4().unwrap();
+
+            // Make sure we can reuse the address and port
+            socket.set_reuseaddr(true).unwrap();
+            socket.set_reuseport(true).unwrap();
+
+            socket.bind(ip).expect("unable to bind to socket");
+            synth_sockets.push(socket);
+        }
+
+        // setup metrics recorder
+        let test_metrics = TestMetrics::default();
+        // clear metrics and register metrics
+        metrics::register_histogram!(METRIC_LATENCY);
+
+        let mut synth_handles = Vec::with_capacity(synth_count);
+        let test_start = tokio::time::Instant::now();
+
+        for socket in synth_sockets {
+            synth_handles.push(tokio::spawn(simulate_peer(node_addr, socket)));
+        }
+
+        // wait for peers to complete
+        for handle in synth_handles {
+            let _ = handle.await;

```

Comment:
I've been wondering if we wouldn't have better performance if we used a [`JoinSet`](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html) instead of a `Vec` to await all these handles :detective:

Reply:
I haven't measured it but solution with JoinSet is more elegant as well as it's dedicated to do this kind of stuff :)

Reply:
oh cool! TIL about this [JoinSet](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html)